jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/BuildReferenceDateUtil.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.common.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+import java.util.Map;
+
+public class BuildReferenceDateUtil {
+  private BuildReferenceDateUtil() { }
+
+  public static void storeReferenceDateInPluginContext(Date now, Map<String, Object> pluginContext, String buildTimestampKey) {
+    pluginContext.put(buildTimestampKey, now);
+  }
+
+  // get a reference date
+  public static Date getBuildReferenceDate(String buildDirectory, String dockerBuildTimestampFile) {
+    // Pick up an existing build date created by build goal previously
+    File tsFile = new File(buildDirectory, dockerBuildTimestampFile);
+    if (!tsFile.exists()) {
+      return new Date();
+    }
+    try {
+      return EnvUtil.loadTimestamp(tsFile);
+    } catch (IOException e) {
+      throw new IllegalStateException("Cannot read timestamp from " + tsFile, e);
+    }
```

Comment:
`EnvUtil.loadTimestamp(tsFile)` takes care of most of this

why aren't we doing something like:
```java
return Optional.ofNullabe(EnvUtil.loadTimestamp(getTimestampFile(...)).orElse(new Date());
```

The logic in this method is repeating the same that we already have in EnvUtil.